### PR TITLE
Added more documentation for better readability

### PR DESCRIPTION
**Pull request by alyssacabading**
Addresses Issue #234442. Added more JSDoc style comments on the src/server-main file to enhance code readability. Implemented documentation on the following parts: the `shouldSpawnCli` conditional, `parseRange` function, and `prompt` function. This is so that there is consistent documentation throughout the file to help future contributors to understand seemingly complex code.

### CODE REVIEW

**Comment by alyssacabading** [comment]
@microsoft-github-policy-service agree

**Review by 818Nawaf — APPROVED** [review]

File: src/server-main.ts [hunk 1]
```diff
@@ -86,8 +89,10 @@ if (shouldSpawnCli) {
 
 	let firstRequest = true;
 	let firstWebSocket = true;
-
 	let address: string | AddressInfo | null = null;
+
+	// Creates an HTTP server that listens for incoming requests
+	// The requests gets processed by the remote extension host agent server
```
Comment (youneshenniwrites) [inline src/server-main.ts:95]:
```suggestion
	// Creates an HTTP server that listens for incoming requests.
        // The requests are processed by the remote extension host agent server.
```

File: src/server-main.ts [hunk 2]
```diff
@@ -96,6 +101,8 @@ if (shouldSpawnCli) {
 		const remoteExtensionHostAgentServer = await getRemoteExtensionHostAgentServer();
 		return remoteExtensionHostAgentServer.handleRequest(req, res);
 	});
+
+	// Listens for when a  client requests to upgrade from HTTP to WebSocket
```
Comment (youneshenniwrites) [inline src/server-main.ts:105]:
```suggestion
	// Listens for when a client requests to upgrade from HTTP to WebSocket
```

**Review by youneshenniwrites — CHANGES_REQUESTED** [review]

**Review by youneshenniwrites — APPROVED** [review]

**Comment by aeschli** [comment]
Thanks you for the PR, but we rather leave the code as is, I think it's clear, what it does without comments

**Pull request closed without merging**